database/sql/src/main/scala/cromwell/database/slick/tables/WorkflowMetadataSummaryEntryComponent.scala

```diff
@@ -63,136 +67,121 @@ trait WorkflowMetadataSummaryEntryComponent {
     }
+    val labelsAndConstraint = NonEmptyList.fromList(labelAndKeyLabelValues.toList.map { case (labelKey, labelValue) => labelExists(labelKey, labelValue) } ).map(AND).toList
 
-    val baseQueryWithIncludeLabels = addLabelsQueries(ands = labelAndKeyLabelValues, ors = labelOrKeyLabelValues, baseWorkflowsQuery = baseQuery)
+    // At least one of the labelOr list of KV pairs must exist:
+    val labelOrConstraint = NonEmptyList.fromList(labelOrKeyLabelValues.toList.map { case (k, v) =>
+      AND(NonEmptyList.of(sql"#$labelsTableCustomName.custom_label_key=$k") :+ sql"#$labelsTableCustomName.custom_label_value=$v")
+    }).map(OR).toList
 
-    if (excludeLabelAndValues.nonEmpty || excludeLabelOrValues.nonEmpty) {
-      // If there are labels to be excluded, run the includes query first then filter anything found in an excludes query.
-      val baseQueryWithExcludeLabels = addLabelsQueries(
-        ands = excludeLabelAndValues,
-        ors = excludeLabelOrValues,
-        baseWorkflowsQuery = baseQuery
-      )
-      baseQueryWithIncludeLabels.filterNot(_.workflowExecutionUuid in baseQueryWithExcludeLabels.map(_.workflowExecutionUuid))
-    } else {
-      baseQueryWithIncludeLabels
+    // *ALL* of the excludeLabelOr list of KV pairs must *NOT* exist:
+    val excludeLabelsOrConstraint = NonEmptyList.fromList(excludeLabelOrValues.toList.map { case (labelKey, labelValue) => NOT(labelExists(labelKey, labelValue)) } ).map(AND).toList
+
+    // At least one of the excludeLabelAnd list of KV pairs must *NOT* exist:
+    val excludeLabelsAndConstraint = NonEmptyList.fromList(excludeLabelAndValues.toList.map { case (labelKey, labelValue) => NOT(labelExists(labelKey, labelValue)) } ).map(OR).toList
+
+    val includeSubworkflowsConstraint = if (includeSubworkflows) List.empty else {
+      val tableName = s"subworkflowMixin" + mixinTableCounter.getAndIncrement()
+      List(sql"""NOT EXISTS(SELECT * from METADATA_ENTRY #$tableName WHERE ((#$tableName.WORKFLOW_EXECUTION_UUID = #$summaryTableAlias.WORKFLOW_EXECUTION_UUID) AND (#$tableName.METADATA_KEY = 'parentWorkflowId') AND (#$tableName.METADATA_VALUE IS NOT NULL)))""")
     }
+
+    val constraintList =
+      excludeLabelsOrLinkingConstraint ++
+        statusConstraint ++
+        nameConstraint ++
+        idConstraint ++
+        submissionTimeConstraint ++
+        startTimeConstraint ++
+        endTimeConstraint ++
+        labelOrConstraint ++
+        labelsAndConstraint ++
+        excludeLabelsOrConstraint ++
+        excludeLabelsAndConstraint ++
+        includeSubworkflowsConstraint
+
+    val where = NonEmptyList.fromList(constraintList) match {
+      case Some(constraints) => List(sql"WHERE ", AND(constraints))
+      case None => List.empty
```

Comment:
if we end up on this side of the match I'm not sure we should go on

Reply:
Spoke in person - if no parameters are provided then the user is asking to return everything. That's not going to happen in the CromIAM world but very much could do in raw Cromwell